### PR TITLE
Signal changes [win32/win64amd]

### PR DESCRIPTION
**Pull request by babsingh**
1) **Properly define MSVC_RUNTIME_DLL in omrsig_internal.hpp**

    Added a fail-safe to properly define MSVC_RUNTIME_DLL if build files
    fail to define MSVC_RUNTIME_DLL. The new CMake build files set
    MSVC_RUNTIME_DLL during compilation. But, the old OMR build files don't
    set MSVC_RUNTIME_DLL during compilation. This fail-safe is to support
    the old build files.

    This is identical to the changes in cmake/modules/OmrMsvcRuntime.cmake.

    Closes: #2904

2) **omrsig should support SIGBREAK on Windows**

    SIGBREAK is a valid signal on Windows. So, omrsig.cpp::validSignalNum
    should accept SIGBREAK as a valid signal on Windows.

3) **Enable signal chaining and use omrsig in omrsignal.c [win32/win64amd]**

    a) Calls to signal are replaced with OMRSIG_SIGNAL. OMRSIG_SIGNAL deals
    with caching application signal handler while registering another OS
    signal handler.

    b) omrsig_handler is called from runHandlers to invoke the application
    signal handler after J9Win*AsyncHandlerRecord->handler is invoked.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

File: omrsigcompat/omrsig_internal.hpp [hunk 1]
```diff
@@ -85,7 +85,30 @@ struct OMR_SigData {
...
+#define MSVC_RUNTIME_DLL "MSVCR110.dll"
+#elif (_MSC_VER == 1800) /* VS 12 */
+#define MSVC_RUNTIME_DLL "MSVCR120.dll"
+#elif (_MSC_VER > 1800) /* VS 14+ */
+#define MSVC_RUNTIME_DLL "UCRTBASE.dll"
+#else /* VS unknown */
+/* Leave MSVC_RUNTIME_DLL undefined to cause a compile error.
+ * This will assure that a user will update MSVC_RUNTIME_DLL
+ * if it is undefined.
+ */
```
Comment (rwy0717) [inline omrsigcompat/omrsig_internal.hpp]: I would do: `#error "unrecognized MSVC_RUNTIME_DLL"`, instead of a comment.

**Comment by rwy0717** [comment]
one nitpick, otherwise looks good.

**Comment by charliegracie** [comment]
The AppVeyor build failed in signal tests. I will launch OMR Win CI build to see if it fails as well.

**Comment by charliegracie** [comment]
@genie-omr build win

**Comment by babsingh** [comment]
@charliegracie these changes should work now.

**Comment by charliegracie** [comment]
Signal test is still failing in the AppVeyor build

**Comment by babsingh** [comment]
@charliegracie AppVeyor build has succeeded.

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by charliegracie** [comment]
I clicked the wrong button on my phone and I didn't mean to merge it yet without full testing... If there are any test failures in the merge build I will have to revert.

**Comment by charliegracie** [comment]
Testing passed on the Push.

**Pull request merged**